### PR TITLE
WA-4771: Merge Bill Payments into Refresh Manager, update Transaction Lists

### PR DESCRIPTION
**Pull request by wgkeller**
## The problem ##

WA-4771:
* Remove local database for transactions
* Implement suspended (coroutine) functions to support API requests
* Upgrade database from v23 -> v24

## The solution ##

What's the technical solution?
* Continued suspended functions for linear style coding for async operations

### CODE REVIEW

File: src/androidTest/kotlin/us/frollo/frollosdk/network/NetworkHelperTest.kt [hunk 1]
```diff
@@ -69,10 +69,4 @@ class NetworkHelperTest {
         val networkHelper = NetworkHelper(appInfo)
         assertNotNull(networkHelper.deviceVersion)
     }
-
-    @Test
```
Comment (VindhyaRajagopal) [inline src/androidTest/kotlin/us/frollo/frollosdk/network/NetworkHelperTest.kt:73]: @wgkeller Why is this test removed? Was this causing any issues

File: src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt [hunk 2]
```diff
@@ -201,8 +197,9 @@ abstract class SDKDatabase : RoomDatabase() {
                     MIGRATION_19_20,
                     MIGRATION_20_21,
                     MIGRATION_21_22,
-                    MIGRATION_22_23
-
+                    MIGRATION_22_23,
+                    MIGRATION_22_23,
```
Comment (VindhyaRajagopal) [inline src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt:201]: MIGRATION_22_23 added twice

File: src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt [hunk 3]
```diff
@@ -875,10 +872,27 @@ abstract class SDKDatabase : RoomDatabase() {
...
+
+        private val MIGRATION_23_24: Migration = object : Migration(23, 24) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                // WARNING: DO NOT USE "BEGIN TRANSACTION" & "COMMIT" as on latest room version
+                // looks like it does it by default. If we add these we will see the error stated in
+                // https://frollo.atlassian.net/browse/WA-3067
+
+                // New changes in this migration:
+                // 1) Drop all transaction caching: WA-4771
+                database.execSQL("DROP TABLE transaction_model")
```
Comment (VindhyaRajagopal) [inline src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt:893]: Do we have to do "DROP INDEX IF EXISTS" of `transaction_model` table explicitly before dropping the table?

**Review by VindhyaRajagopal — CHANGES_REQUESTED** [review]

File: gradle.properties [hunk 4]
```diff
@@ -12,12 +12,4 @@
...
-
-# When configured, Gradle will run in incubating parallel mode.
-# This option should only be used with decoupled projects. More details, visit
-# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
-# org.gradle.parallel=true
-
-# Disabling full R8 mode for now as its causing issues with gradle 7.0
-# For now we are using R8 Compatibility mode
-#android.enableR8.fullMode=true
+org.gradle.daemon=false
```
Comment (VindhyaRajagopal) [inline gradle.properties:15]: Also, please bump the SDK version to 3.25.0.
And, add these changes summary in changelog.md file for 3.25.0 release.

**Review by VindhyaRajagopal — CHANGES_REQUESTED** [review]

File: src/androidTest/kotlin/us/frollo/frollosdk/network/NetworkHelperTest.kt (hunk 1 above)
Comment (wgkeller) [reply src/androidTest/kotlin/us/frollo/frollosdk/network/NetworkHelperTest.kt:73]: A field was removed from NetworkHelper (userAgent) - which is no longer required. As per code comments:
` // "us.frollo.frollosdk|SDK1.0.0|B777|Android8.1.0|API2.0"
    @Deprecated("Not needed anymore by the Host on v2 APIs")
    internal val userAgent: String
        get() = "${BuildConfig.LIBRARY_PACKAGE_NAME}|SDK${BuildConfig.SDK_VERSION_NAME}|B${BuildConfig.SDK_VERSION_CODE}|Android${Build.VERSION.RELEASE}|API$API_VERSION"
`

File: src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt (hunk 2 above)
Comment (wgkeller) [reply src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt:201]: Merge issue, I'll fix

File: src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt (hunk 3 above)
Comment (wgkeller) [reply src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt:896]: Fixed with latest commit

File: src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt (hunk 2 above)
Comment (wgkeller) [reply src/main/kotlin/us/frollo/frollosdk/database/SDKDatabase.kt]: Fixed with latest commit

File: gradle.properties (hunk 4 above)
Comment (wgkeller) [reply gradle.properties:15]: Fixed with latest commit

**Review by VindhyaRajagopal — APPROVED** [review]

**Pull request merged**